turn_alerts/views.py

```diff
@@ -0,0 +1,128 @@
+)
+
+
+class TurnAlertsLayerView(generics.GenericAPIView):
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request, *args, **kwargs):
+        try:
+            webhook_type = request.headers["X-Turn-Hook-Subscription"]
+        except KeyError:
+            return Response(
+                {"X-Turn-Hook-Subscription": ["This header is required."]},
+                status.HTTP_400_BAD_REQUEST,
+            )
+        org_id = kwargs["org_id"]
+
+        try:
+            organization = Organization.objects.get(id=org_id)
+            turn_alerts = TurnActions.objects.filter(org=organization)
+        except Organization.DoesNotExist:
+            return JsonResponse(data={}, status=status.HTTP_400_BAD_REQUEST)
+
+        on_fallback_channel = request.headers.get("X-Turn-Fallback-Channel", "0") == "1"
+        is_turn_event = request.headers.get("X-Turn-Event", "0") == "1"
+
+        for alert in turn_alerts:
+            journey_id = alert.journey_id
+            org_error_code = alert.error_code
+            engage_token = organization.engage_token
+            engage_url = organization.engage_url
+
+        if webhook_type == "whatsapp" or is_turn_event:
+            WhatsAppWebhookSerializer(data=request.data).is_valid(raise_exception=True)
+            for inbound in request.data.get("messages", []):
+
+                message_type = inbound.pop("type")
+                direction = inbound["_vnd"]["v1"]["direction"]
+                message_requests_total.labels(
+                    fallback_channel=on_fallback_channel,
+                    direction=direction,
+                    message_type=message_type,
+                ).inc()
+            for statuses in request.data.get("statuses", []):
+                message_status = statuses.get("status")
+
+                if "errors" in statuses:
+                    recipient_id = statuses.get("recipient_id")
+                    error_code = statuses["errors"][0].get("code")
+                    event_count.labels(
+                        message_status=message_status,
+                        error_code=error_code,
+                        conversation_id=None,
+                        conversation_type=None,
+                    ).inc()
+                    if error_code == org_error_code:
+                        match = re.match(
+                            (
+                                r"^\s*(?:\+?(\d{1,3}))?[-. (]*(\d{3})[-. )]"
+                                r"*(\d{3})[-. ]*(\d{4})(?: *x(\d+))?\s*$"
+                            ),
+                            recipient_id,
+                        )
+                        if match:
+                            start_turn_journey(
+                                recipient_id, journey_id, engage_url, engage_token
+                            )
+
+                else:
+                    recipient_id = statuses.get("recipient_id")
```

Comment:
Are we using this?

Reply:
No, only need for the falied event that starts the journey.